src/handshake/server.rs

```diff
@@ -141,24 +179,48 @@ impl<S: Read + Write, C: Callback> HandshakeRole for ServerHandshake<S, C> {
                 if !tail.is_empty() {
                     return Err(Error::Protocol("Junk after client request".into()))
                 }
-                let extra_headers = {
-                    if let Some(callback) = self.callback.take() {
-                        callback.on_request(&result)?
```

Comment:
But the error in the previous version also was returned, right? So the biggest difference is that the type of the error has been changed? And so does the handling, i.e. instead of returning an error, the function will rreturn `ProcessingResult::Continue` and write some response to the incoming request, correct?

Reply:
The previous version had wrong error type. It used the same `Error` as the library itself which is nonsense. Returning an error from the callback resulted in immediate connection drop and returning the same error to the library user.